30_regexp_lab/relab.txt

```diff
@@ -1,17 +1,35 @@
 Use grep to print only the lines from "input" for which:
 
-1. There is a binary number in the line
+
+1. There is a binary number in the line
 
-2. There is an even binary number in the line
+2. There is an even binary number in the line
 
-3. Line includes a hexadecimal number
+
 
-4. Line starts with a lowercase letter
+3. Line includes a hexadecimal number
 
+
+
+4. Line starts with a lowercase letter
+
+
+cat input | grep '^[a-z]'
+
+
 5. Line is not empty
 
-6. Line starts with a lowercase letter and ends in a number
+
+cat input | grep -v '^$'

```

Comment:
Great. I prefer the shorter `grep .` (because `.` is any character, so a line that has ANY character is certainly not empty)